src/Symfony/Component/Console/Command/InvokableCommand.php

```diff
@@ -188,4 +192,61 @@ public function handleSignal(int $signal, int|false $previousExitCode = 0): int|
     {
         return $this->signalableCommand?->handleSignal($signal, $previousExitCode) ?? false;
     }
+
+    public function isInteractive(): bool
+    {
+        if (null === $this->interactions) {
+            $this->collectInteractions();
+        }
+
+        return [] !== $this->interactions;
+    }
+
+    public function interact(InputInterface $input, OutputInterface $output): void
+    {
+        if (null === $this->interactions) {
+            $this->collectInteractions();
+        }
+
+        foreach ($this->interactions as $interaction) {
+            $interaction->interact($input, $output, $this->getParameters(...));
+        }
+    }
+
+    private function collectInteractions(): void
+    {
+        $invokableThis = $this->invokable->getClosureThis();
+
+        $this->interactions = [];
+        foreach ($this->invokable->getParameters() as $parameter) {
+            if ($spec = Argument::tryFrom($parameter)) {
```

Comment:
This duplicates the call to `(Argument|Input)::tryFrom`. This should be kept in memory?

Also, why not adding the `InteractiveQuestion` for options? They can be required (yes).

Reply:
Actually, no.. options can't be required by design. That's the reason I've excluded them from this automatic feature, since enforcing them as required on every execution would go against their intended use.

The same applies to optional arguments: only required arguments are prompted automatically. However, you still have full control through the interact method to implement any custom logic you need.